libraries/Parsers/Microsoft.Bot.Builder.Parsers.LU/Parser/Diagnostic.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+using Antlr4.Runtime;
+
+namespace Microsoft.Bot.Builder.Parsers.LU.Parser
+{
+    public enum DiagnosticSeverity
+    {
+        /// <summary>Raise an error</summary>
+        Error,
+
+        /// <summary>Raise warning instead of error</summary>
+        Warn
+    }
+
+    public class Diagnostic : Error
+    {
+        public Diagnostic(Range range, string message, string severity = "ERROR")
```

Comment:
Do we want a discrete set of severities or allow any string? If the former, then we should validate severity 3 lines below. And also define compile-time constants or enums.

Reply:
I see you defined an enum and that the factory methd BuildDiagnostic already expects an instance of the enum. So could you hep me understand why we expect a string here and hardcode to a fixed string instead of reusing the enum? I looked a bit for a reason but couldn't find :)

Reply:
Solved